test-end-to-end/distribution/ConcurrencyE2E.java

```diff
@@ -146,4 +155,101 @@ public void concurrentInsertionOfDuplicateAttributes_doesNotCreateGhostVertices(
         assertEquals(10, numOfSurnames);
         assertEquals(10, numOfAges);
     }
+
+    private <T extends Element> void insertElements(GraknClient.Session session, List<T> elements,
+                                                    int threads, int insertsPerCommit) throws ExecutionException, InterruptedException {
+        ExecutorService executorService = Executors.newFixedThreadPool(threads);
+        int listSize = elements.size();
+        int listChunk = listSize / threads + 1;
+
+        List<CompletableFuture<Void>> asyncInsertions = new ArrayList<>();
+        for (int threadNo = 0; threadNo < threads; threadNo++) {
+            boolean lastChunk = threadNo == threads - 1;
+            final int startIndex = threadNo * listChunk;
+            int endIndex = (threadNo + 1) * listChunk;
+            if (endIndex > listSize && lastChunk) endIndex = listSize;
+
+            List<T> subList = elements.subList(startIndex, endIndex);
+            System.out.println("indices: " + startIndex + "-" + endIndex + " , size: " + subList.size());
+            CompletableFuture<Void> asyncInsert = CompletableFuture.supplyAsync(() -> {
+                long start2 = System.currentTimeMillis();
+                GraknClient.Transaction tx = session.transaction().write();
+                int inserted = 0;
+                for (int elementId = 0; elementId < subList.size(); elementId++) {
+                    T element = subList.get(elementId);
+                    GraqlInsert insert = Graql.insert(element.patternise(Graql.var("x" + elementId).var()).statements());
+                    tx.execute(insert);
+                    if (inserted % insertsPerCommit == 0) {
+                        tx.commit();
+                        inserted = 0;
+                        tx = session.transaction().write();
+                    }
+                    inserted++;
+                }
+                tx.commit();
+                System.out.println("Thread: " + Thread.currentThread().getId() + " elements: " + subList.size() + " time: " + (System.currentTimeMillis() - start2));
+                return null;
+            }, executorService);
+            asyncInsertions.add(asyncInsert);
+        }
+        CompletableFuture.allOf(asyncInsertions.toArray(new CompletableFuture[]{})).get();
+        executorService.shutdown();
+    }
+
+    private static String generateString(int length) {
+        final boolean useLetters = true;
+        final boolean useNumbers = true;
+        return RandomStringUtils.random(length, useLetters, useNumbers);
```

Comment:
this method can just be `return RandomStringUtils.random(length, true, true);`, also no need to make random `final` variables

Reply:
yes, but then we don't know what the flags are for, no? It makes it more explicit

Reply:
it's all one click away, but ok it makes sense, but really no need to be final